chain/types/ethtypes/eth_transactions.go

```diff
@@ -21,8 +19,35 @@ import (
 	"github.com/filecoin-project/lotus/chain/types"
 )
 
-const Eip1559TxType = 2
+const (
+	// LegacyHomesteadEthTxSignaturePrefix defines the prefix byte used to identify the signature of a legacy Homestead Ethereum transaction.
+	LegacyHomesteadEthTxSignaturePrefix = 0x01
+	Eip1559TxType                       = 0x2
 
+	ethLegacyHomesteadTxType    = 0x0
+	ethLegacyHomesteadTxChainID = 0x0
+)
+
+// EthTransaction defines the interface for Ethereum-like transactions.
+// It provides methods to convert transactions to various formats,
+// retrieve transaction details, and manipulate transaction signatures.
+type EthTransaction interface {
+	Sender() (address.Address, error)
+	Signature() (*typescrypto.Signature, error)
+	InitialiseSignature(sig typescrypto.Signature) error
```

Comment:
I do love a good english spelling code interface, I wonder if this on a public interface will cause problems for our American friends?

Reply:
Eh should be fine. They can lookup the Oxford dictionary :)